### PR TITLE
recipes changes for hipblas, rocprim, rocfft, rocsolver for rocm3.7.0

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocsolver/package.py [hunk 1]
```diff
@@ -14,15 +14,21 @@ class Rocsolver(CMakePackage):
...
 
+    depends_on('cmake@3:', type='build')
+    depends_on('numactl', when='@3.7.0')
+    depends_on('hsa-rocr-dev@3.7.0', type='build', when='@3.7.0')
+
+    for ver in ['3.5.0', '3.7.0']:
+        depends_on('hip@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
+        depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('rocblas@' + ver, type='build', when='@' + ver)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocsolver/package.py]: type='build' for rocblas? :thinking:

File: var/spack/repos/builtin/packages/rocfft/package.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ class Rocfft(CMakePackage):
         depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
         depends_on('hip@' + ver, when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('comgr@' + ver, when='@' + ver)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocfft/package.py]: `comgr` is repeated?

File: var/spack/repos/builtin/packages/hipblas/package.py [hunk 3]
```diff
@@ -14,14 +14,15 @@ class Hipblas(CMakePackage):
...
-    depends_on('rocsolver')
-    depends_on('rocblas')
-    depends_on('rocm-device-libs', type='build')
-    depends_on('comgr', type='build')
-
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+    for ver in ['3.5.0', '3.7.0']:
+        depends_on('hip@' + ver, when='@' + ver)
+        depends_on('rocsolver@' + ver, type='build', when='@' + ver)
+        depends_on('rocblas@' + ver, type='build', when='@' + ver)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/hipblas/package.py]: type='build'?

File: var/spack/repos/builtin/packages/rocfft/package.py (hunk 2 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/rocfft/package.py]: it requires link dependency too.. Modified accordingly

File: var/spack/repos/builtin/packages/hipblas/package.py (hunk 3 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/hipblas/package.py]: made changes as per your comments

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@srekolam if you're confused by Spack's dependency types, https://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types was just updated to be a bit more clear. If you still have any questions about the difference between these types, let me know.

@haampie does this look good to you now?

**Comment by srekolam** [comment]
@adamjstewart , can you please let me know if there is anything pending on my side.

**Pull request merged**